esx_service/vmdk_ops.py

```diff
@@ -702,6 +705,24 @@ def disk_attach(vmdk_path, vm):
     kv_status_attached, kv_uuid, attach_mode = getStatusAttached(vmdk_path)
     logging.info("Attaching {0} as {1}".format(vmdk_path, attach_mode))
 
+    # If the volume is attached then check if the attach is stale, if so detach
+    # the volume from the VM its attached to and attach to the requesting VM.
+    if kv_status_attached and kv_uuid != vm.config.uuid:
+       attached_to_vm = findVmByUuid(kv_uuid)
```

Comment:
Need log message if it fails. And cleanup of KV file in this case is needed too.